engine/src/cubos/engine/plugins/renderer.cpp

```diff
@@ -0,0 +1,57 @@
+#include <cubos/engine/plugins/renderer.hpp>
+#include <cubos/engine/plugins/transform.hpp>
+#include <cubos/engine/plugins/window.hpp>
+
+#include <cubos/engine/ecs/draw_system.hpp>
+#include <components/cubos/camera.hpp>
+#include <components/cubos/grid.hpp>
+
+#include <cubos/engine/gl/deferred/renderer.hpp>
+#include <cubos/engine/gl/renderer.hpp>
+#include <cubos/engine/gl/frame.hpp>
+
+#include <cubos/core/gl/camera.hpp>
+
+#include <cubos/core/settings.hpp>
+
+static void startup(gl::Renderer& renderer, const cubos::core::io::Window& window)
```

Comment:
This is using a blank settings object. You should use the existing one:
```suggestion
static void startup(gl::Renderer& renderer, const cubos::core::io::Window& window, const cubos::core::Settings& settings)
{
    auto& renderDevice = window->getRenderDevice();
    renderer =
        std::make_shared<gl::deferred::Renderer>(renderDevice, window->getFramebufferSize(), settings);
}
```